source/extensions/formatter/cel/cel.cc

```diff
@@ -0,0 +1,93 @@
+#include "source/extensions/formatter/cel/cel.h"
+
+#include <sys/stat.h>
+
+#include <string>
+
+#include "source/common/config/metadata.h"
+#include "source/common/formatter/substitution_formatter.h"
+#include "source/common/http/utility.h"
+#include "source/common/protobuf/utility.h"
+#include "source/common/runtime/runtime_features.h"
+
+constexpr absl::string_view DefaultUnspecifiedValueString = "-";
+
+#if defined(USE_CEL_PARSER)
+#include "parser/parser.h"
+#endif
+
+namespace Envoy {
+namespace Extensions {
+namespace Formatter {
+
+namespace {
+
+std::string truncate(std::string str, absl::optional<size_t> max_length) {
```

Comment:
It's better to avoid calling this function if max_length is not set. Otherwise, you're copying strings.